### PR TITLE
Home Work #3 Different Types Of Functions

### PR DESCRIPTION
**Pull request by AndreyTests**

### CODE REVIEW

File: functions/task1.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// function declaration
+function squareSurface ( width, height ) {
```
Comment (sttaran) [inline functions/task1.js]: function name should be an action - "what to do". Example - `calculateArea`

File: functions/task2.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+function isAdult ( age ) {
```
Comment (sttaran) [inline functions/task2.js]: you can name it isAdult but I would consider rename the function so the would represent the action

File: functions/task3.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const counter = ( num ) => {
```
Comment (sttaran) [inline functions/task3.js]: function name should be an action like `countDown`

File: functions/task3.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    console.log( num )
+    if (num <= 0) {
```
Comment (sttaran) [inline functions/task3.js]: Фунція повинна викликатися, доки num не стане менше або рівне 0.

function should not be called when num equals 0

File: functions/task4.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+function divide ( numerator, denominator ) {
+        try {
+            if ( typeof numerator !== 'number' || typeof denominator !== 'number' ) {
```
Comment (sttaran) [inline functions/task4.js:3]: you can make it easier 
```js
function divide ( numerator, denominator ) {
    try {
        if ( typeof numerator !== 'number' || typeof denominator !== 'number' ) {
            throw new Error('All arguments should be a number');
        }
        
        if ( denominator === 0 ) {
            throw new Error('Impossible to divide on 0');
        }
            return numerator / denominator;

    } catch ( error ) {
        return error.message;
    } finally {
        console.log ( 'Робота завершена' );
    }
}
```

File: functions/task4.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+                throw new Error ( 'All arguments should be a number' );
+            } else if ( denominator === 0 ) {
+                throw new Error ( 'Impossible to divide on 0' );
+            } else {
+                let result = numerator / denominator;
+                return result;
+            }
+        } catch ( error ) {
+            return error.message;
```
Comment (sttaran) [inline functions/task4.js]: Створіть блок catch в якому виведіть повідомлення про помилку в консоль.

print error but not return it as result from function

you can return null in this case

**Review by sttaran — CHANGES_REQUESTED** [review]

File: functions/task1.js (hunk 1 above)
Comment (AndreyTests) [reply functions/task1.js]: Updated the names of function.

File: functions/task2.js (hunk 2 above)
Comment (AndreyTests) [reply functions/task2.js]: Updated the names of function.

File: functions/task3.js (hunk 3 above)
Comment (AndreyTests) [reply functions/task3.js]: Updated the names of function.

File: functions/task3.js (hunk 4 above)
Comment (AndreyTests) [reply functions/task3.js]: Changing the condition.

File: functions/task4.js (hunk 5 above)
Comment (AndreyTests) [reply functions/task4.js:3]: Now the code is easier.

File: functions/task4.js (hunk 6 above)
Comment (AndreyTests) [reply functions/task4.js]: Updating of block "catch".

**Comment by AndreyTests** [comment]
@sttaran, please check I fixed all issues/comments

**Pull request merged**